src/config/backends/extlinux-file.ts

```diff
@@ -0,0 +1,86 @@
+import * as _ from 'lodash';
+
+import { ConfigOptions } from './backend';
+import { ExtLinuxParseError } from '../../lib/errors';
+
+export interface ExtlinuxFile {
+	labels: Label;
+	globals: Directive;
+}
+
+export interface Label {
+	[labelName: string]: Directive;
+}
+
+export interface Directive {
+	[directive: string]: string;
+}
+
+export abstract class ConfigurableDirective {
+	// Parses the values for this directive
+	public abstract parse(directives: Directive): ConfigOptions;
+	// Return the value to be set for this directive using the provided ConfigOptions
+	public abstract value(opts: ConfigOptions, existingValue: string): string;
+}
+
+export class AppendDirective extends ConfigurableDirective {
+	public parse(directives: Directive): ConfigOptions {
+		const appendEntry = directives.APPEND;
+
+		if (appendEntry == null) {
+			throw new ExtLinuxParseError(
+				'Could not find APPEND directive in default extlinux.conf boot entry',
+			);
+		}
+
+		const configOpts: ConfigOptions = {};
+		const values = appendEntry.split(' ');
+
+		for (const value of values) {
+			const parts = value.split('=');
+			if (parts.length !== 2) {
```

Comment:
Did you look at whether the values could contain a `=` too? How is that handled. Given we're testing for example 2 entries, that could cause issues. I expect they're just not allowed, but it's worth checking.

Reply:
To my knowledge a key within the APPEND value cannot have more then one =.

APPEND ro rootwait isolcpu=2,3 other=4 // OK
APPEND ro rootwait isolcpu=2,3other=4 // WRONG

Therefore your comment has made me aware of an issue where the parse wouldn't error on these invalid values. I have modified it to catch any config options that have more then a KEY and VALUE.

```js
const [KEY, VALUE, ...more] = appendValue.split('=');
if (more.length > 0) {
// APPEND value is not formatted correctly
}
```